infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/SimpleLeafNode.java

```diff
@@ -57,6 +58,19 @@ public Bytes32 hashTreeRoot() {
     return cachedHash;
   }
 
+  @Override
+  public void updateDigestWithHashTreeRoot(MessageDigest messageDigest) {
+    Bytes32 cachedHash = this.cachedHash;
+    if (cachedHash != null) {
+      cachedHash.update(messageDigest);
+      return;
+    }
+    data.update(messageDigest);
+    if (data.size() < MAX_BYTE_SIZE) {
+      ZERO_LEAVES[MAX_BYTE_SIZE - data.size()].getData().update(messageDigest);
+    }
```

Comment:
Do you think we still actually need `cachedHash` with this update? 
I believe the `SimpleLeafNode.hashTreeRoot()` is only called by the parent branch node `hashTreeRoot()`. 
Could it be the case that `cachedHash` doesn't add perf improvements anymore while still occupies memory?